### ISSUE TITLE
[Bug]: When using partition to insert data, the ETCD storage space suddenly increases

### ISSUE THREAD

**Issue by nall95** [milvus-io/milvus#24817]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.5
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka): pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2): java v2.2.1
- OS(Ubuntu or CentOS): CentOS
```

### Current Behavior

When using partition to insert data, the ETCD storage space suddenly increases.
The partition number is 1000+.

The etcd curve is shown below.
![微信图片_20230612110304](https://github.com/milvus-io/milvus/assets/37284023/f5b36b81-04f1-48e2-8a19-d97277ca4ec6)

When we stop using partitions to insert data, the etcd storage space start recycling.

![微信图片_20230612110639](https://github.com/milvus-io/milvus/assets/37284023/e675e978-961d-488b-b6e9-ea9149f58752)

We use the command below to insert data.
![微信图片_20230612110932](https://github.com/milvus-io/milvus/assets/37284023/74a9e338-9acf-4bb8-8489-416e25d6d771)

### Milvus Log

milvus log:
[fbe2a3f4579da7857d91242240d8f1f1_3ebeb31d4b3f66827f5ca8742090301d_8.gz](https://github.com/milvus-io/milvus/files/11717137/fbe2a3f4579da7857d91242240d8f1f1_3ebeb31d4b3f66827f5ca8742090301d_8.gz)

etcd log:
[f328bc5a4c72191f791827dcc7587296_8e52bc39ee8c244fcb3271b02cbf19f9_8.gz](https://github.com/milvus-io/milvus/files/11717149/f328bc5a4c72191f791827dcc7587296_8e52bc39ee8c244fcb3271b02cbf19f9_8.gz)

birdwatcher:
[cf37eb493052ebbae626da3c0fd5d11e_f3de74ba1cc06a93281b7989f5a01aa4_8.gz](https://github.com/milvus-io/milvus/files/11717152/cf37eb493052ebbae626da3c0fd5d11e_f3de74ba1cc06a93281b7989f5a01aa4_8.gz)

**Comment by yanliang567** [comment]
sounds like a known issue. 
@nall95 could you please share in what kinds of scenarios you have 1000+ partitions?
/assign @nall95 

/assign @LoveEachDay 
could you please help to take a look?

**Comment by LoveEachDay** [comment]
@nall95 Yes, this is working in progress. We'll refine the etcd data usage for large partitions scenarios. Checkout pr: https://github.com/milvus-io/milvus/pull/24819

Etcd is a mvcc meta store which will store multiple versions for the same key. Based on the above `etcd db size` figure, you'd change the backend quota size to 8G. And also enable auto compaction, which will remove old reversion, and the db usage will be decreased.

**Comment by xiaofan-luan** [comment]
This is a known issue and should be fixed by 2.2.12

**Comment by yanliang567** [comment]
/assign @jingkl 
could you please help to verify the issue on 2.2.12?

/unassign
